File: plugins/drsommer/drsommer.go
package drsommer

import (
	"fmt"
	"regexp"

	"github.com/abourget/slick"
	"github.com/fsr/matthias/util/random"
)

var drsommerChannel string

type drsommer struct{}

func init() {
	slick.RegisterPlugin(&drsommer{})
}

var robot *slick.Bot

// InitPlugin ...
func (drsommer *drsommer) InitPlugin(bot *slick.Bot) {
	var conf struct {
		DrSommer struct {
			Channel string
		}
	}

	bot.LoadConfig(&conf)

	drsommerChannel = conf.DrSommer.Channel

	bot.Listen(&slick.Listener{
		Matches:            regexp.MustCompile("(?i)^(?:hey |hallo |lieber )?dr\\.? sommer,?"),
		MessageHandlerFunc: drsommer.chatHandler,
	})

	robot = bot
}

func (drsommer *drsommer) chatHandler(listen *slick.Listener, msg *slick.Message) {
	question := fmt.Sprintf("\"%s\" - %s (%d)", msg.Text, random.StringFromList(names), random.IntBetween(10, 15))
	robot.SendToChannel(drsommerChannel, question)
}

var names = []string{
	"Shantalle",
	"Schackeline",
	"Tamara",
	"Cindy",
	"Jenny",
	"Clair-Gina",
	"Justin",
	"Jerome",
	"Chantal",
	"Jacqueline",
	"Yves",
	"Sylvia",
	"Claudia",
	"Dörte",
}

File: bot/DrSommer.hs
-- Description:
--   Dr. Sommer

-- Dependencies:
--   None

-- Configuration:
--   None

-- Commands:
--   Dr. Sommer, ich hab da mal ne Frage - Stelle eine anonyme Frage in #dr_sommer

-- Author:
--   kiliankoe
module DrSommer where

import Marvin.Prelude
import Data.Text (replace)


script :: IsAdapter a => ScriptInit a
script = defineScript "drsommer" $ do
    hear (r [CaseInsensitive] "^(hey |hallo |lieber )?dr\\.? sommer,?") $ do
        msg <- getMessage
        name <- randomFrom names
        i <- randomValFromRange (10, 15)
        messageRoom "#dr_sommer" $ toStrict $ format "\"{}\" - {} ({})" (content msg, name, i :: Int)

    -- help matthias respond to priv messages as well
    respond (r [CaseInsensitive] "(hey |hallo |lieber )?dr\\.? sommer,?") $ do
        msg <- getMessage
        let question = replace (content msg) "matthias " ""
        name <- randomFrom names
        i <- randomValFromRange (10, 15)
        messageRoom "#dr_sommer" $ toStrict $ format "\"#{}\" - {} ({})" (question, name, i :: Int)


names :: [Text]
names = [
    "Shantalle",
    "Schackeline",
    "Tamara",
    "Cindy",
    "Jenny",
    "Clair-Gina",
    "Justin",
    "Jerome",
    "Chantal",
    "Jacqueline",
    "Yves",
    "Sylvia",
    "Claudia",
    "Dörte"
    ]
